src/EditorInput.re

```diff
@@ -451,7 +467,36 @@ module Make = (Config: {
     };
 
   let keyUp = (~context, ~key, bindings) => {
-    handleKeyCore(~context, Up(key), {...bindings, suppressText: false});
+    let pressedScancodes =
+      IntSet.remove(key.scancode, bindings.pressedScancodes);
+    let bindings = {...bindings, suppressText: false, pressedScancodes};
+
+    // If everything has been released, fire an [AllKeysReleased] event,
+    // in case anything is listening for it.
+    let initialEffects =
+      if (IntSet.is_empty(pressedScancodes)) {
+        let releaseBindings =
+          bindings.bindings
+          |> List.filter_map(applyKeyToBinding(~context, AllKeysReleased));
+
+        let effect =
+          switch (releaseBindings) {
+          | [hd, ..._] =>
+            switch (hd.action) {
+            | Dispatch(payload) => [Execute(payload)]
+            | _ => []
+            }
+          | [] => []
+          };
```

Comment:
You could simplify this by nesting the patterns:

```suggestion
          switch (releaseBindings) {
          | [{action: Dispatch(payload), _}, ..._] => Execute(payload)]
          | _ => []
          };
```

Reply:
Thanks @glennsl ! That is much nicer